src/ResourceManager/Consumption/Commands.Consumption/Cmdlets/UsageDetails/GetAzureRmConsumptionUsageDetail.cs

```diff
@@ -29,121 +29,187 @@ public class GetAzureRmConsumptionUsageDetail : AzureConsumptionCmdletBase
     {
         const int MaxNumberToFetch = 1000;
 
-        [Parameter(Mandatory = true, HelpMessage = "Name of a specific invoice to get the usage details that associate with.", ParameterSetName = Constants.ParameterSetNames.InvoiceItemParameterSet)]
-        [ValidateNotNullOrEmpty]
-        public string InvoiceName { get; set; }
-
-        [Parameter(Mandatory = true, HelpMessage = "Name of a specific billing period to get the usage details that associate with.", ParameterSetName = Constants.ParameterSetNames.BillingPeriodItemParameterSet)]
+        [Parameter(Mandatory = false, HelpMessage = "Name of a specific billing period to get the usage details that associate with.")]
         [ValidateNotNullOrEmpty]
         public string BillingPeriodName { get; set; }
 
-        [Parameter(Mandatory = false, HelpMessage = "Determine the maximum number of records to return.")]
+        [Parameter(Mandatory = false, HelpMessage = "Expand the usages based on MeterDetails, or AdditionalProperties.")]
         [ValidateNotNull]
-        public int? MaxCount { get; set; }
+        public string Expand { get; set; }
 
         [Parameter(Mandatory = false, HelpMessage = "Include meter details in the usages.")]
         public SwitchParameter IncludeMeterDetails { get; set; }
 
         [Parameter(Mandatory = false, HelpMessage = "Include additional properties in the usages.")]
         public SwitchParameter IncludeAdditionalProperties { get; set; }
 
-        [Parameter(Mandatory = false, HelpMessage = "The start date (in UTC) of the usages.")]
+        [Parameter(Mandatory = false, HelpMessage = "The start date (in UTC) of the usages to filter.")]
         [ValidateNotNull]
         public DateTime? StartDate { get; set; }
 
-        [Parameter(Mandatory = false, HelpMessage = "The end date (in UTC) of the usages.")]
+        [Parameter(Mandatory = false, HelpMessage = "The end date (in UTC) of the usages to filter.")]
         [ValidateNotNull]
         public DateTime? EndDate { get; set; }
 
+        [Parameter(Mandatory = false, HelpMessage = "The resource group of the usages to filter.")]
+        [ValidateNotNull]
+        public string ResourceGroup { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The instance name of the usages to filter.")]
+        [ValidateNotNull]
+        public string InstanceName { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The instance id of the usages to filter.")]
+        [ValidateNotNull]
+        public string InstanceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The tag of the usages to filter.")]
+        [ValidateNotNull]
+        public string Tags { get; set; }
```

Comment:
Rename to "Tag" (powershell guidelines)

Reply:
Tags is renamed as Tag, and .md file and recorded sessions are regenerated accordingly.